src/libstd/sys/unix/process/process_fuchsia.rs

```diff
@@ -0,0 +1,190 @@
+        // Create a launchpad
+        mx_cvt(launchpad_create(job_copy, self.get_argv()[0],
+                                &mut launchpad as *mut *mut launchpad_t))?;
+        // Set the process argv
+        mx_cvt(launchpad_arguments(launchpad, self.get_argv().len() as i32 - 1,
+                                   self.get_argv().as_ptr()))?;
+        // Setup the environment vars
+        mx_cvt(launchpad_environ(launchpad, envp))?;
+        mx_cvt(launchpad_add_vdso_vmo(launchpad))?;
+        mx_cvt(launchpad_clone_mxio_root(launchpad))?;
+        // Load the executable
+        mx_cvt(launchpad_elf_load(launchpad, launchpad_vmo_from_file(self.get_argv()[0])))?;
+        mx_cvt(launchpad_load_vdso(launchpad, MX_HANDLE_INVALID))?;
+        mx_cvt(launchpad_clone_mxio_cwd(launchpad))?;
+
+        // Clone stdin, stdout, and stderr
+        if let Some(fd) = stdio.stdin.fd() {
+            launchpad_transfer_fd(launchpad, fd, 0);
+        } else {
+            launchpad_clone_fd(launchpad, 0, 0);
+        }
+        if let Some(fd) = stdio.stdout.fd() {
+            launchpad_transfer_fd(launchpad, fd, 1);
+        } else {
+            launchpad_clone_fd(launchpad, 1, 1);
+        }
+        if let Some(fd) = stdio.stderr.fd() {
+            launchpad_transfer_fd(launchpad, fd, 2);
+        } else {
+            launchpad_clone_fd(launchpad, 2, 2);
+        }
+
+        // We don't want FileDesc::drop to be called on any stdio. It would close their fds. The
+        // fds will be closed once the child process finishes.
+        mem::forget(stdio);
+
+        for callback in self.get_closures().iter_mut() {
+            callback()?;
+        }
+
+        let process_handle = mx_cvt(launchpad_start(launchpad))?;
+
+        // Successfully started the launchpad, so launchpad_destroy shouldn't get called
+        mem::forget(launchpad_destructor);
+
+        Ok((launchpad, process_handle))
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Processes
+////////////////////////////////////////////////////////////////////////////////
+
+pub struct Process {
+    launchpad: *mut launchpad_t,
+    handle: mx_handle_t,
+    status: Option<ExitStatus>,
+}
+
+impl Process {
+    pub fn id(&self) -> u32 {
+        self.handle as u32
+    }
+
+    pub fn kill(&mut self) -> io::Result<()> {
+        use sys::magenta::*;
+
+        // If we've already waited on this process then the pid can be recycled
+        // and used for another process, and we probably shouldn't be killing
+        // random processes, so just return an error.
+        if self.status.is_some() {
+            Err(io::Error::new(io::ErrorKind::InvalidInput,
+                           "invalid argument: can't kill an exited process"))
+        } else {
+            unsafe {
+                mx_cvt(mx_handle_close(self.handle))?;
+                launchpad_destroy(self.launchpad);
```

Comment:
If this is a "proper handle" as opposed to a pid, it may be best to do this in a `Drop` implementation (like Windows) perhaps?